redhat/squeezeboxserver.service

```diff
@@ -22,6 +22,13 @@ Environment="SQUEEZEBOX_ADDITIONAL_ARGS="
 # in /etc/systemd/system/logitechmediaserver.service.d/*.conf
 EnvironmentFile=-/etc/sysconfig/squeezeboxserver
 
+# Do not change this, it will most likely break the squeezeboxserver
+# installation. The directory /usr/lib/perl5/vendor_perl is where the RPM
+# package drops the Slim perl modules. Setting PERL5LIB to this location will
+# make sure that the squeezebox server executable looks here before it scans the
+# directories in the perl @INC array.
+Environment="PERL5LIB=/usr/lib/perl5/vendor_perl"
```

Comment:
see above

Reply:
Hi,

I actually did think about this too. The setting of this variable the way I have done will only have an impact for the process running squeezeboxserver, it will not change the value globally. But even so, it will of course have an impact if someone have set PERL5LIB with the direct intent to add or change something to the squeezeboxserver process

It is absolutely no problem at all to change the SYSV script to use:

export PERL5LIB="/usr/lib/perl5/vendor_perl;$PERL5LIB"

For the systemd unit file it is, unfortunately more complex. By default the systemd unit files will not pick up any environment variables (the systems I work with in my day job only pick up PATH and LANG in systemd). It is thus not possible to have a statement like:

Environment="PERL5LIB=/usr/lib/perl5/vendor_perl;$PERL5LIB"

in the unit file. It will just evaluate to /usr/lib/perl5/vendor_perl. This is also the reason why the SQUEEZEBOX_ARGS variable from the file /etc/sysconfig/squeezeboxserver can't be used in the systemd unit file.

If someone has set the PERL5LIB variable in order to change something for squeezeboxserver, then they will have to use a drop-in file in /etc/systemd/system/squeezeboxserver.service.d. This is, unfortunately, how systemd is designed.

The systemd is in many ways different to how people are used to how Linux works. Another example is that systemd does not respect the ulimits set in /etc/security/limits.conf. Instead systemd uses it own default values. These default values can be changed system wide by editing /etc/systemd/system.conf. This can actually lead to some strange effects in some specific cases if default values are changed for a unit running as a specific user, but the values in the normal limits.conf for that user are left unchanged.

To summarise,

- SYSV init file: I am happy to change the init file. How do I do that? By creating a new pull request?
- systemd unit file: I wouldn't know how to do this.

Please let me know what you think.

Regards, Johan

Reply:
Thanks for the explanation! Sounds good to me.